src/Resolution.ts

```diff
@@ -96,16 +99,16 @@ export default class Resolution {
    * Creates a resolution instance with configured provider
    * @param provider - any provider with sendAsync function impelmented
    */
-  static provider(provider: Web3Provider): Resolution {
-    return new this({ blockchain: { web3Provider: provider } });
+  static provider(provider: Web3Provider, providerType: ProviderType): Resolution {
+    return new this({ blockchain: { web3Provider: provider, providerType: providerType } });
   }
 
   /**
    * Creates a resolution instance from configured jsonRPCProvider
    * @param provider - any jsonRPCprovider will work as long as it's prototype has send(method, params): Promise<any> method
    */
-  static jsonRPCprovider(provider): Resolution {
-    return new this({ blockchain: { web3Provider: provider.send } });
+  static jsonRPCprovider(provider, providerType: ProviderType): Resolution {
```

Comment:
`providerType` should have the default value `"ethers"`, so we don't break backwards compatibility